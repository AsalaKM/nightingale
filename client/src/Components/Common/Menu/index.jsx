import React, { Component } from "react";
import axios from "axios";
import { withRouter } from "react-router-dom";

import {
  MenuDiv,
  Menu,
  MenuItem,
  MainDiv,
  MenuClose,
  MenuIcon
} from "./index.style";

class HumburgerMenu extends Component {
  state = {
    toggleShow: false
  };

  onClick = () => {
    const { toggleShow } = this.state;
    this.setState({ toggleShow: !toggleShow });
  };

  handleLogout = () => {
    axios.post("./api/user/logout").then(() => {
      this.props.handleLogout();
      this.props.history.push("/");
    });
  };

  render() {
    const { toggleShow } = this.state;
    return (
      <>
        <MenuDiv>
          {!toggleShow && (
            <MenuIcon className="fas fa-bars" onClick={this.onClick} />
          )}
          {toggleShow && (
            <Menu>
              <MenuClose className="fas fa-times" onClick={this.onClick} />
              <MainDiv>
                <MenuItem to="/dashboard" onClick={this.onClick}>
                  <i className="fas fa-smile-beam" /> Dashboard
                </MenuItem>
                <MenuItem to="/conversations" onClick={this.onClick}>
                  <i className="fas fa-comments" /> Conversation
                </MenuItem>
                <MenuItem to="/chat" onClick={this.onClick}>
                  <i className="fas fa-comment" /> Start Chat
                </MenuItem>
                <MenuItem to="/advice" onClick={this.onClick}>
                  <i className="fas fa-link" /> Advice
                </MenuItem>
                <MenuItem onClick={this.handleLogout} as="p">
                  <i className="fas fa-sign-out-alt" />
                  LogOut
                </MenuItem>
              </MainDiv>
            </Menu>
          )}
        </MenuDiv>
      </>
    );
  }
}
export default withRouter(HumburgerMenu);
