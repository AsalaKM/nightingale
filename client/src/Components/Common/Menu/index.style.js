import styled from "styled-components";
import { Link } from "react-router-dom";

export const MenuDiv = styled.div`
  position: absolute;
  top: 0;
  right: 0;
  padding: 20px 20px 0px 0px;
  z-index: 1000;
`;
export const Menu = styled.div`
  position: absolute;
  background: #e3e3ea;
  z-index: 100;
  top: 0;
  right: 0;
  width: 250px;
  height: 100vh;
  transition: all 500ms ease;
  padding: 40px 10px;
`;
export const MainDiv = styled.div`
  padding: 2rem 0 0 2rem;
`;
export const MenuItem = styled(Link)`
  margin-bottom: 2rem;
  display: flex;
  align-items: center;
  color: #6b6277;
  font-weight: bold;
  letter-spacing: 1px;
  transition: all 500ms ease;
  cursor: pointer;

  i {
    margin-right: 0.5rem;
  }

  :hover {
    text-indent: 10px;
    color: #925de5;
  }
`;
export const MenuIcon = styled.i`
  color: #925de5;
  font-size: 25px;
  cursor: pointer;
`;
export const MenuClose = styled.i`
  position: absolute;
  font-size: 23px;
  top: 0;
  left: 0;
  margin: 20px 0px 10px 10px;
  cursor: pointer;
  color: #6b6277;
  cursor: pointer;
  :hover {
    color: #925de5;
  }
`;
