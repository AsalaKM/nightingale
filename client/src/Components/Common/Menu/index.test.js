import React from "react";
import renderer from "react-test-renderer";
import Menu from "./index.jsx";

describe("Menu", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<Menu />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
