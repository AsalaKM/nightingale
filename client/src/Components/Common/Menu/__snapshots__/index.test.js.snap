// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu renders correctly 1`] = `
<div
  className="sc-bdVaJa fbaCFo"
>
  <i
    className="fas fa-bars sc-ifAKCX eEdOhH"
    onClick={[Function]}
  />
</div>
`;
